Dockerfile

```diff
@@ -22,78 +22,80 @@
 # 7. copy ALL of REMOTE_SOURCES to REMOTE_SOURCES_DIR, not just upstream1/cachito.env and upstream1/app/registry-ca.pem
 # 8. add Brew metadata
 
-# Stage 1 - Install dependencies
+# Stage 1 - Build nodejs skeleton
 #@follow_tag(registry.redhat.io/ubi9/nodejs-18:1)
-FROM registry.access.redhat.com/ubi9/nodejs-18:1 AS deps
+FROM registry.access.redhat.com/ubi9/nodejs-18:1 AS skeleton
 USER 0
 
+# Install isolate-vm dependencies
+RUN dnf update -y && \
+  dnf install -y zlib-devel brotli-devel && \
+  dnf clean all
+
 # Env vars
 ENV YARN=./.yarn/releases/yarn-1.22.19.cjs
 ENV REMOTE_SOURCES=.
 ENV REMOTE_SOURCES_DIR=/opt/app-root/src
 
 WORKDIR $REMOTE_SOURCES_DIR/
-COPY $REMOTE_SOURCES $REMOTE_SOURCES_DIR
-RUN chmod +x $REMOTE_SOURCES_DIR/.yarn/releases/yarn-1.22.19.cjs
+COPY $REMOTE_SOURCES/.yarn $REMOTE_SOURCES_DIR/.yarn
```

Comment:
IIRC bad things happen when you only copy SOME files over... cachito is fickle and if its config files are missing, things jsut don't work. Better to copy everything in one go, and if you realllllly need to, delete the things you know you don't need/want.

Reply:
This was just a skeleton layer that the other stages use as a base image. We copy the other files over in the subsequent stages so that we get better caching. (ex: when you change some code without changing the dependencies, it wouldn't need to reinstall the dependencies)